x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/exceptions/is_valid_exceptions_list.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ListArray } from '@kbn/securitysolution-io-ts-list-types';
+import { ExceptionListTypeEnum } from '@kbn/securitysolution-io-ts-list-types';
+import type { RulesClient } from '@kbn/alerting-plugin/server';
+import { findRules } from '../search/find_rules';
+
+/**
+ * Util to check if linked exceptions exist already the system
+ * @param exceptionLists {array} - exception lists and items to import
+ * @returns {Promise} exception lists or error if more than one default list found
+ */
+export const isValidExceptionList = async ({
+  exceptionsList,
+  rulesClient,
+  ruleId,
+}: {
+  exceptionsList: ListArray | undefined;
+  ruleId: string | undefined;
+  rulesClient: RulesClient;
+}): Promise<boolean> => {
+  if (!exceptionsList) {
+    return true;
+  }
+
+  const newDefaultExceptionsLists = exceptionsList.filter(
+    (list) => list.type === ExceptionListTypeEnum.RULE_DEFAULT
+  );
+
+  if (newDefaultExceptionsLists.length === 0) return true;
+
+  if (newDefaultExceptionsLists.length > 1) {
+    throw new Error('More than one default exception list found on rule');
```

Comment:
We're throwing here but not catching this error or using the message passed as param. We could just return `false` and then the logic in the controllers is still valid:
```
        const isExceptionListValid = await isValidExceptionList({
          exceptionsList: params.exceptions_list,
          rulesClient,
          ruleId: params.id,
        });
        if (!isExceptionListValid) {
          return siemResponse.error({
            statusCode: 409,
            body: `default exception list already exists`,
          });
        }
```

Reply:
As suggested by @xcrzx , I updated it to throw errors for all invalid cases in this function.
As I understand it's already a way how some validation perform, and each request has try/catch to process this error